### PR TITLE
Add `clamp` tests

### PR DESCRIPTION
**Pull request by zoddicus**
<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
PTAL, this is ready for review

File: src/webgpu/shader/execution/builtin/clamp.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const cfg: Config = t.params;
+    cfg.cmpFloats = correctlyRoundedThreshold();
+
+    // This array must be strictly increasing, since that ordering determines
+    // the expected values.
+    const test_values: Array<Scalar> = [
+      i32Bits(0x80000000),
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/builtin/clamp.spec.ts:111]: Any reason you're not using the constants for int-min here?

File: src/webgpu/shader/execution/builtin/clamp.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      i32(-2),
+      i32(-1),
+      i32(0),
+      i32(1),
+      i32(2),
+      i32Bits(0x70000000),
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/builtin/clamp.spec.ts:117]: Maybe also add an i32-max case? (`0x7fffffff`)

**Review by ben-clayton — APPROVED** [review]
Two nits around the `i32` cases, but looks great!

File: src/webgpu/shader/execution/builtin/clamp.spec.ts (hunk 1 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/builtin/clamp.spec.ts]: Done

File: src/webgpu/shader/execution/builtin/clamp.spec.ts (hunk 2 above)
Comment (zoddicus) [reply src/webgpu/shader/execution/builtin/clamp.spec.ts:117]: Done

**Pull request merged**